### ISSUE TITLE
False positive in empty_enum_arguments rule

### ISSUE THREAD

**Issue by ZevEisenberg** [realm/SwiftLint#1597]
Unrelated to #1549.

file:

```swift
import Foundation

let myString = "FOO"

switch myString {
case "foo".uppercased(): // ← this triggers
    print("shouty")
default:
    break
}
```

```sh
bash-3.2$ swiftlint version
0.19.0
bash-3.2$ swiftlint lint --path empty_enum_arguments.swift 
Linting Swift files at path empty_enum_arguments.swift
Linting 'empty_enum_arguments.swift' (1/1)
/Users/zev/Desktop/empty_enum_arguments.swift:6:22: warning: Empty Enum Arguments Violation: Arguments can be omitted when matching enums with associated types if they are not used. (empty_enum_arguments)
Done linting! Found 1 violation, 0 serious in 1 file.
```
